consensus/core/src/block.rs

```diff
@@ -294,13 +300,44 @@ pub(crate) struct SignedBlock {
 impl SignedBlock {
     // TODO: add verification.
 
-    // TODO: will refactor once the signing approach has been introduced.
-    pub(crate) fn new(block: Block) -> Self {
+    /// Should only be used when constructing the genesis blocks
+    pub(crate) fn new_genesis(block: Block) -> Self {
         Self {
             inner: block,
             signature: Bytes::default(),
         }
     }
+    pub(crate) fn new<S>(block: Block, signer: &S) -> ConsensusResult<Self>
```

Comment:
We could be even more specific here and just ask for a `NetworkKeyPair` as a signer if we do find our selfs using any mocks etc.